### PR TITLE
Replacing parameters with localparams

### PR DESCRIPTION
**Pull request by dpetrisko**
Some tools (genus) and flows (ILM) like to have all parameters in name to resolve, otherwise subtle differences in instantiation can cause name mismatches. This causes all default parameters to be included in the name and makes the name extremely long and complex. For instance, the manycore tile actually breaks the filename length limit for Unix.

This PR replaces "localparameters" with localparam. In BP this also helped compile times significantly, but that's a secondary benefit. @taylor-bsg I don't think there are any tools we care about that can't handle localparams in module headers. Icarus Verilog is the only one I would guess, and there's no way the rest of bsg_manycore is going through that anyway

### CODE REVIEW

**Comment by taylor-bsg** [comment]
Interesting -- how things have changed! So localparam in arguments is supported by DC, VCS, Surelog, Verilator, and Vivado?

**Comment by dpetrisko** [comment]
And genus/xcelium ;)

BP uses them heavily, so if they weren’t supported it would have come up for anything we run BP through

**Comment by taylor-bsg** [comment]
I think rather than placing lots of parameter keywords in many places, you can just use the parameter keyword in the next parameter after you add a localparam, to switch it back?

**Comment by dpetrisko** [comment]
I like the explicitness of having the parameter keyword but it’s up to you. The error is very unintuitive, so not sure it’s worth the clarity hit for future maintainers

**Comment by dpetrisko** [comment]
Think this is good to go once CI passes

File: v/bsg_cache_dma_to_wormhole.v
Comment (tommydcjung) [inline v/bsg_cache_dma_to_wormhole.v:16]: missing "parameter" on the first ones.

File: v/bsg_manycore_hor_io_router.v
Comment (tommydcjung) [inline v/bsg_manycore_hor_io_router.v:33]: This is not localparam

**Review by tommydcjung — CHANGES_REQUESTED** [review]
Left feedback that must be addressed before merging.

**Comment by taylor-bsg** [comment]
Yes, let's remove the redundant localparam and parameters. Unfortunately, it is things like this that kill our code size metrics versus other alternative HDL's (something that is particularly relevant for BlackParrot)

File: v/bsg_cache_dma_to_wormhole.v
Comment (dpetrisko) [reply v/bsg_cache_dma_to_wormhole.v]: fixed

File: v/bsg_manycore_hor_io_router.v
Comment (dpetrisko) [reply v/bsg_manycore_hor_io_router.v]: good catch, this file isn't compiled normally so it wasn't caught

File: v/bsg_manycore_accel_default.v
Comment (tommydcjung) [inline v/bsg_manycore_accel_default.v:7]: There are all these first parameter in each module missing the keyword "parameter"

Comment (dpetrisko) [reply v/bsg_manycore_accel_default.v:7]: technically, the first one isn't required because the list starts as parameter by default

Comment (taylor-bsg) [reply v/bsg_manycore_accel_default.v]: I think it's okay to omit the first one. But can we go ahead and remove the redundant parameter and localparam statements? I still see a bunch in the code

Comment (tommydcjung) [reply v/bsg_manycore_accel_default.v]: then we can remove parameter on the second one? Or put the parameter on the first one to be more clear.

Comment (dpetrisko) [reply v/bsg_manycore_accel_default.v]: I just changed the parameter to localparam so no net change in keywords. If we want to code golf, we could have all parameters and then all localparams? Then it’s only 1 “localparam” per module

Comment (dpetrisko) [reply v/bsg_manycore_accel_default.v]: @taylor-bsg Just pushed a version that's minified without grouping. I think anything further should be its own PR

File: v/bsg_manycore_tile_compute_array_ruche.v
Comment (tommydcjung) [inline v/bsg_manycore_tile_compute_array_ruche.v:65]: Can you also fix y/x_cord_width_p, pod_y/x_cord_width_p in this file?

**Comment by tommydcjung** [comment]
these files are deprecated and no longer in use. These should be either reverted or deleted.
```
v/bsg_ruche_anti_buffer.v
v/bsg_manycore_rocc_wrapper.v
v/bsg_manycore_rocc_streambuf_wrapper.v
v/bsg_manycore_packet_streamer.v
v/bsg_manycore_mesh.v
v/bsg_manycore_links_to_fsb.v
v/bsg_manycore_link_to_rocc.v
```

**Review by tommydcjung — APPROVED** [review]

File: v/bsg_manycore_tile_compute_array_ruche.v
Comment (dpetrisko) [reply v/bsg_manycore_tile_compute_array_ruche.v:65]: fixed!

**Comment by dpetrisko** [comment]
>these files are deprecated and no longer in use. These should be either reverted or deleted.

Reverted and added #609

**Pull request merged**
